library/ssl_srv.c

```diff
@@ -3585,15 +3585,19 @@ static int ssl_parse_certificate_verify( mbedtls_ssl_context *ssl )
     /* Read the message without adding it to the checksum */
     do {
 
-        if( ( ret = mbedtls_ssl_read_record_layer( ssl ) ) != 0 )
+        do ret = mbedtls_ssl_read_record_layer( ssl );
+        while( ret == MBEDTLS_ERR_SSL_CONTINUE_PROCESSING );
```

Comment:
@RonEld Yes exactly, for that particular part of the code the changes are meant to only change the way the looping is done, but essentially leaving the code as it was before. The main change is when we introduce looping where we previously returned `WANT_READ`. 
Thanks for looking at the code, and let me know if there are potential issues in the other parts of the PR.